lib/index.js

```diff
@@ -64,130 +64,95 @@ function transformJs(filepath, src) {
         );
 }
 
+const checkPaths = (paths, requestedFilePath) => {
+    for (const pathPrefix in paths) {
+        // if the requested file does not start with the path prefix we can move on
+        if (!requestedFilePath.startsWith(pathPrefix)) {
+            continue;
+        }
+        // strip the path prefix from the requested path and create the resolved path
+        const servedPath = paths[pathPrefix];
+        const strippedPath = requestedFilePath.slice(pathPrefix.length);
+        const resolvedPath = path.join(servedPath, strippedPath);
+        // if the resolved path exists and is not a directory return it
+        if (fs.existsSync(resolvedPath)) {
+            if (fs.lstatSync(resolvedPath).isDirectory()) {
+                continue;
+            }
+            return resolvedPath;
+        }
+    }
+    return false;
+};
+
 /**
- * A Karma middleware factory which uses the config.basePath to configure module resolution.
+ * Attempst to serve a response based on the given paths from which to serve files, and a requested path.
  *
- * @param {*} basePath - the basePath configuration from karma
- * @param {string[]} indexFiles - an array of index files to attempt to resolve directories to, in order of preference, defaults to `['index.html', 'index.js']`
```

Comment:
It was removed, its not needed anymore, since we can now use the middleware later in the chain and serve just the things we need, it no longer conflicts with other index serving middlewares.

Reply:
👍